.github/workflows/lib_compat.h

```diff
@@ -0,0 +1,18 @@
+// This header forces the use of a particular version of glibc symbols.
+// This can make the resulting binary portable to systems with older glibcs.
+//
+// It must be included in each TU, with CFLAGS="-include glibc_compat.h" etc.
+//
+// We only list symbols known to be used. This is paired with a test enforcing
+// the max glibc symbol version. If the test fails, the list should be extended.
+// Find the old alternatives version to target with e.g.:
+//   objdump -T /lib/x86_64-linux-gnu/lib{c,m,pthread,rt}.so.* | grep "\bexpf\b"
+
+#define FORCE_SYMBOL_VERSION(sym, version) \
+  __asm__(".symver " #sym "," #sym "@" #version)
+
+FORCE_SYMBOL_VERSION(expf, GLIBC_2.2.5);
```

Comment:
nit: maybe provide a wrapper to not repeat `GLIBC_2.2.5` all the time ?

Reply:
It's not always 2.2.5 in general, though it *usually* is (I think some dynamic linker change in that version meant it included new versions of all symbols). See https://github.com/wheybags/glibc_version_header/blob/master/version_headers/x64/force_link_glibc_2.18.h

I figured the least error-prone thing was to include the version exactly as it was in the objdump command, so it can be copy/pasted and to make the connection with the test clearer too.